plugins/container/go-worker/worker.go

```diff
@@ -48,13 +48,17 @@ func workerLoop(ctx context.Context, cb asyncCb, containerEngines []container.En
 	}
 
 	for {
-		chosen, val, _ := reflect.Select(cases)
+		chosen, val, recvOk := reflect.Select(cases)
```

Comment:
In case we fail to recv the value, it means that the channel has been closed, and thus the worker goroutine at idx `chosen` has left. Drop it from our cases slice.